grommet/grommet
# Restore arrow key functionality to Menu component

#### What does this PR do?

This PR addresses missing [WAI-ARIA keyboard interaction](https://www.w3.org/TR/wai-aria-practices/#menu) handling which was removed in #3464 to fix onClick handling in FireFox. Up and Down arrow keys can be used to navigate between menu-items after they are in focus. This also raises the testing coverage for the Menu component by ~1.5%.

#### Where should the reviewer start?

`src/js/components/Menu/Menu.js`

#### What testing has been done on this PR?

I implemented test cases to assert that the arrow keys function properly when being used. I also tested on Chrome, Firefox, Safari, Opera, and Edge to make sure that the bug which was fixed in #3464 had not returned when this code was restored. All browsers I tested with handled onClick properly.

#### How should this be manually tested?

 Run `yarn test` to ensure that all tests pass and view coverage. Run `yarn storybook` to ensure that keypress handling works properly.

#### Any background context you want to provide?

See https://github.com/grommet/grommet/pull/4580#issuecomment-705089240
In the future, toggling the menu should automatically set the focus to the first menu element.

#### What are the relevant issues?

#### Screenshots (if appropriate)

Improved test coverage (from #4580)
<img width="1661" alt="Screen Shot 2020-10-07 at 2 44 38 PM" src="https://user-images.githubusercontent.com/24704789/95392267-738b7c00-08c6-11eb-9ffa-1f2f9f7e5c42.png">

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

#### Is this change backwards compatible or is it a breaking change?

Backwards Compatible
